lib/compass/functions/_lists.scss

```diff
@@ -78,13 +78,24 @@
 }
 
 @if not(function-exists(compact)) {
-  @function compact($vars...) {
+  @function compact($args...) {
+    $first: nth($args, 1);
+    $sep: comma;
     $list: ();
-    @each $var in $vars {
-        @if $var {
-            $list: append($list, $var, comma);
-        }
+
+    @if length($args) == 1 and type_of($first) == 'list' {
+      $args: $first;
+      $sep: list-separator($args);
+    }
+
+    @for $index from 1 through length($args) {
```

Comment:
`@each $arg in $args` won't work here when the first argument is `list`. It only returns first value in the first list. 
eg: 
`((opacity .2s ease), false, false, false)`

The $arg will be `opacity` if we use `@each`